src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -234,13 +235,94 @@ public AndroidTzData()
+            // tzlookup.xml is an autogenerated file that contains timezone ids in this form:
+            //
+            // <timezones ianaversion="2019b">
+            //   <countryzones>
+            //     <country code="au" default="Australia/Sydney" everutc="n">
+            //       <id alts="Australia/ACT,Australia/Canberra,Australia/NSW">Australia/Sydney</id>
+            //       ...
+            //       ...
+            //       <id>Australia/Eucla</id>
+            //     </country>
+            //     <country ...>
+            //       ...
+            //       ...
+            //       ...
+            //     </country>
+            //   </countryzones>
+            // </timezones>
+            //
+            // Once the timezone cache is populated with the IDs, we reference tzlookup id tags and
+            // remove IDs and their associated information (byteoffsets and lengths) from the current
+            // AndroidTzData instance.
+            private void FilterBackwardIDs(string lookupPath)
+            {
+                try
+                {
+                    using (StreamReader sr = File.OpenText(lookupPath))
+                    {
+                        var tzLookupIDs = new HashSet<string>();
+                        string? tzLookupLine;
+                        while (sr.Peek() >= 0)
+                        {
+                            if (!(tzLookupLine = sr.ReadLine())!.TrimStart().StartsWith("<id"))
+                                continue;
+
+                            int idStart = tzLookupLine.IndexOf('>') + 1;
+                            int idLength = tzLookupLine.LastIndexOf("</") - idStart;
+                            if (idStart == 0 || idLength < 0)
+                            {
+                                // Either the start tag <id ... > or the end tag </id> are not found
+                                continue;
+                            }
+                            string id = tzLookupLine.Substring(idStart, idLength);
+                            tzLookupIDs.Add(id);
+                        }
+                        var cleanIDs = new List<string>();
+                        var cleanOffsets = new List<int>();
+                        var cleanLengths = new List<int>();
+                        for (int i = 0; i < _ids.GetLength(0); ++i)
+                        {
+                            if (tzLookupIDs.Contains(_ids[i]))
+                            {
+                                cleanIDs.Add(_ids[i]);
+                                cleanOffsets.Add(_byteOffsets[i]);
+                                cleanLengths.Add(_lengths[i]);
+                            }
+                        }
+                        _ids = cleanIDs.ToArray();
+                        _byteOffsets = cleanOffsets.ToArray();
+                        _lengths = cleanLengths.ToArray();
+                    }
+                }
+                catch (FileNotFoundException)
```

Comment:
>catch [](http://example.com/codeflow?start=16&length=5)

we should catch all exceptions and then ignore the filtering all together without any throwing. Getting the duplicated display names would be better than throwing and causing app failures

Reply:
Ignore the filtering altogether, as in should we not even try to filter if `GetSystemTImeZones` is called a second time?
Should we log a warning here, and what is the standard way for logging that in CoreLib?

Reply:
> Ignore the filtering altogether, as in should we not even try to filter if GetSystemTImeZones is called a second time?

Yes. We try the filtering once and cache the result to use for any subsequent calls. If filtering fails, ignore it moving forward and don't try it again. It is unlikely to succeed if it fail first time. 

> Should we log a warning here, and what is the standard way for logging that in CoreLib?

I don't think we log that. Maybe we can have assert in the debug builds instead.

Reply:
One other note, we have the API TimeZoneInfo.ClearCachedData which clears all cached data. If this is called and we need to recreate the cache, we should do the filtering at that time.

Reply:
The filtering is now based on an `_isBackwards` boolean array that is created and set when the IDs are populated. I believe the filtering still works after calling TimeZoneInfo.ClearCachedData

Reply:
> I believe the filtering still works after calling TimeZoneInfo.ClearCachedData

do you mean you recreate the `_isBackwards` array again or you are saying `_isBackwards` is created only once and used even after clearing the cache? 

Here is a scenario you can validate:

-  Call FindSystemTimeZoneUsingId("America/Los_Angeles")
- Call GetSystemTimeZones
- Validate the behavior that backward zones are filtered
- Clear the cache
- Call GetSystemTimeZones
- Validate the behavior one more time that backward zones are filtered

Reply:
Its created whenever `_ids` are created (and `_byteOffsets` and `_lengths`), 
Using 
```
TimeZoneInfo LA = TimeZoneInfo.FindSystemTimeZoneById("America/Los_Angeles");
Console.WriteLine($"LA DisplayName: {LA.DisplayName}");
ReadOnlyCollection<TimeZoneInfo> tzCollection = TimeZoneInfo.GetSystemTimeZones();
HashSet<String> tzDisplayNames = new HashSet<String>();
foreach (TimeZoneInfo timezone in tzCollection)
{
    tzDisplayNames.Add(timezone.DisplayName);
}
Console.WriteLine($"{tzCollection.Count == tzDisplayNames.Count} tzCollection: {tzCollection.Count} tzDisplayNames: {tzDisplayNames.Count}");
TimeZoneInfo.ClearCachedData();
tzCollection = TimeZoneInfo.GetSystemTimeZones();
tzDisplayNames = new HashSet<String>();

foreach (TimeZoneInfo timezone in tzCollection)
{
    tzDisplayNames.Add(timezone.DisplayName);
}
Console.WriteLine($"{tzCollection.Count == tzDisplayNames.Count} tzCollection: {tzCollection.Count} tzDisplayNames: {tzDisplayNames.Count}");
```
Both evaluate to true